File: web/src/services/sys/sysPermissionController.ts
// @ts-ignore
/* eslint-disable */
import { request } from '@umijs/max';

/** 此处后端没有提供注释 POST /api/sys/permission/create */
export async function createSysPermission(
  body: API.SysPermissionCreateInput,
  options?: { [key: string]: any },
) {
  return request<API.RSysPermissionDetailView>('/api/sys/permission/create', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 DELETE /api/sys/permission/delete/${param0} */
export async function deleteSysPermissionById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.deleteSysPermissionByIdParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/sys/permission/delete/${param0}`, {
    method: 'DELETE',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/permission/id/${param0} */
export async function findSysPermissionById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysPermissionByIdParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RSysPermissionDetailView>(`/api/sys/permission/id/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/permission/list */
export async function findSysPermissionList(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysPermissionListParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysPermissionPageView>('/api/sys/permission/list', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/permission/page */
export async function findSysPermissionPage(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysPermissionPageParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysPermissionPageView>('/api/sys/permission/page', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
      pageable: undefined,
      ...params['pageable'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/permission/tree */
export async function findSysPermissionTree(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysPermissionTreeParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysPermissionTreeView>('/api/sys/permission/tree', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/permission/tree-root */
export async function findSysPermissionTreeRoot(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysPermissionTreeRootParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysPermissionTreeRootView>('/api/sys/permission/tree-root', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /api/sys/permission/update */
export async function updateSysPermissionById(
  body: API.SysPermissionUpdateInput,
  options?: { [key: string]: any },
) {
  return request<API.RSysPermissionDetailView>('/api/sys/permission/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: src/main/kotlin/com/zhengchalei/xadmin/modules/sys/controller/SysPermissionController.kt
/*
Copyright 2024 [郑查磊]

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
package com.zhengchalei.xadmin.modules.sys.controller

import com.zhengchalei.xadmin.commons.QueryPage
import com.zhengchalei.xadmin.commons.R
import com.zhengchalei.xadmin.modules.sys.domain.Log
import com.zhengchalei.xadmin.modules.sys.domain.OperationType
import com.zhengchalei.xadmin.modules.sys.domain.dto.*
import com.zhengchalei.xadmin.modules.sys.service.SysPermissionService
import jakarta.validation.constraints.NotNull
import org.babyfish.jimmer.client.meta.Api
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@Api("sys")
@Validated
@RestController
@RequestMapping("/api/sys/permission")
class SysPermissionController(val sysPermissionService: SysPermissionService) {
    @PreAuthorize("hasAuthority('sys:permission:read') or hasAnyRole('admin')")
    @GetMapping("/id/{id}")
    fun findSysPermissionById(@PathVariable id: Long): R<SysPermissionDetailView> {
        val data = sysPermissionService.findSysPermissionById(id)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:permission:read') or hasAnyRole('admin')")
    @GetMapping("/list")
    fun findSysPermissionList(@NotNull specification: SysPermissionListSpecification): R<List<SysPermissionPageView>> {
        val data = sysPermissionService.findSysPermissionList(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:permission:read') or hasAnyRole('admin')")
    @GetMapping("/tree-root")
    fun findSysPermissionTreeRoot(specification: SysPermissionListSpecification): R<List<SysPermissionTreeRootView>> {
        val data = sysPermissionService.findSysPermissionTreeRoot(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:permission:read') or hasAnyRole('admin')")
    @GetMapping("/tree")
    fun findSysPermissionTree(specification: SysPermissionListSpecification): R<List<SysPermissionTreeView>> {
        val data = sysPermissionService.findSysPermissionTree(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:permission:read') or hasAnyRole('admin')")
    @GetMapping("/tree-select")
    fun findSysPermissionTreeSelect(
        specification: SysPermissionListSpecification
    ): R<List<SysPermissionTreeSelectView>> {
        val data = sysPermissionService.findSysPermissionTreeSelect(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:permission:read') or hasAnyRole('admin')")
    @GetMapping("/page")
    fun findSysPermissionPage(
        @NotNull specification: SysPermissionListSpecification,
        @NotNull pageable: QueryPage,
    ): R<MutableList<SysPermissionPageView>> {
        val data = sysPermissionService.findSysPermissionPage(specification, pageable.page())
        return R.success(data)
    }

    @Log(value = "创建系统权限", type = OperationType.CREATE)
    @PreAuthorize("hasAuthority('sys:permission:create') or hasAnyRole('admin')")
    @PostMapping("/create")
    fun createSysPermission(
        @NotNull @RequestBody sysPermissionCreateInput: SysPermissionCreateInput
    ): R<SysPermissionDetailView> {
        val data = sysPermissionService.createSysPermission(sysPermissionCreateInput)
        return R(data = data)
    }

    @Log(value = "修改系统权限", type = OperationType.UPDATE)
    @PreAuthorize("hasAuthority('sys:permission:edit') or hasAnyRole('admin')")
    @PostMapping("/update")
    fun updateSysPermissionById(
        @NotNull @RequestBody sysPermissionUpdateInput: SysPermissionUpdateInput
    ): R<SysPermissionDetailView> {
        val data = sysPermissionService.updateSysPermissionById(sysPermissionUpdateInput)
        return R(data = data)
    }

    @Log(value = "删除系统权限", type = OperationType.DELETE)
    @PreAuthorize("hasAuthority('sys:permission:delete') or hasAnyRole('admin')")
    @DeleteMapping("/delete/{id}")
    fun deleteSysPermissionById(@NotNull @PathVariable id: Long): R<Boolean> {
        sysPermissionService.deleteSysPermissionById(id)
        return R()
    }
}
